cf/samples/javascript-samples/geotab/geotab-to-sap-cloud.js

```diff
@@ -0,0 +1,768 @@
+
+main();
+
+// Only function declarations after this point.
+
+/**
+ * The main function.
+ */
+async function main() {
+    SAP_DeviceManagement = await new Swagger(SAP_Swagger_API, {
+        authorizations: {
+            basicAuth: {
+                username: config.SAP_username,
+                password: config.SAP_password
+            }
+        }
+    });
+    //console.log('SAP northbound API', SAP_DeviceManagement);
+    //console.log('Apis:', SAP_DeviceManagement.apis);
+    //console.log('Definitions:', SAP_DeviceManagement.spec.definitions);
+
+    sensorTypeId_LogRecord = await createType_LogRecord();
+    sensorTypeId_StatusData = await createType_StatusData();
+
+    gatewayId = await findCloudGatewayRest();
+    routerDevice = await initRouterDevice();
+    
+    // Storage is used to preserve the state of the polling between the invocations
+    // of the script. If the storage is removed from disk of the script is started
+    // from a different location, the same data will be copied from Geotab to SAP
+    // twice, resulting in duplicates in SAP IoT Service.
+    await storage.init();
+    
+    // Data of different types are polled and POSTed in parallel.
+    await Promise.all([
+        copy_LogRecords(),
+        copy_StatusData()
+    ]);
+}
+
+/**
+ * Read the configuration file in JSON format synchronously, parse the content and return the result.
+ * @param {string} filename a file name relative to the script directory
+ * @return {Object}  the parsed configuration file
+ */
+function readConfigFile(filename) {
+    const abs_path = path.join(__dirname, filename);
+    const json = fs.readFileSync(abs_path, {
+        'encoding': 'utf8'
+    });
+    return JSON.parse(json);
+}
+
+
+/**
+ * Create the SAP router device unless it already exists and fetch its certificate.
+ * @return {Object} A proxy object with postMeasurementData() asynchronous method.
+ */
+async function initRouterDevice() {
+    const routerDeviceId = await createDeviceUnlessExists(gatewayId, 'router', true);
+    const absPathPem = path.join(__dirname, config.SAP_router_device_pem_filename);
+    const absPathPass = path.join(__dirname, config.SAP_router_device_pass_filename);
+    if (!(fs.existsSync(absPathPem) && fs.existsSync(absPathPass))) {
+        const resp = await SAP_DeviceManagement.apis.Devices.getPEMCertificateUsingGET({
+            tenantId: config.SAP_tenantID,
+            deviceId: routerDeviceId
+        });
+        // console.debug(resp);
+        if (resp.ok) {
+            fs.writeFileSync(absPathPem, resp.body.pem);
+            fs.writeFileSync(absPathPass, resp.body.secret);
+            console.info('Saved the PEM and the passphrase for the router device.');
+        }
+    }
+
+    const router_device_pem = fs.readFileSync(absPathPem, {
+        'encoding': 'utf8'
+    }); // contains both the certificate and the private key
```

Comment:
as general suggestion (personal opinion) put the comment always on a new line because it could break the line formatting (related to format max font per line)